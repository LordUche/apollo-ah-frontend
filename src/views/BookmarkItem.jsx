import React from 'react';
import { Link } from 'react-router-dom';

/**
 * @description - returns a single bookmark item card
 * @return {JSX}
 */

function BookmarkItem(props) {
  const {
    category,
    image,
    articleTitle,
    articleDescription,
    authorsName,
    authorsImage,
    articleSlug,
    readTime,
  } = props;

  return (
    <div className="col-lg-4 col-sm-6">
      <Link to={`/article/${articleSlug}`} className="link-surface" />
      <div className="article transition">
        <div className="article-cover">
          <img alt="" src={image} className="article-cover-img transition" />
        </div>
        <div className="article-item">
          <span className="article-category">{category}</span>
          <h4>{articleTitle}</h4>
          <p>{articleDescription}</p>
          <div className="article-author">
            <img alt="" src={authorsImage} />
            <div>
              <h6>{authorsName}</h6>
              <span className="article-author-dot" />
              <span>{readTime}</span>
              <i className="fas fa-bookmark transition bookmarked" />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default BookmarkItem;
