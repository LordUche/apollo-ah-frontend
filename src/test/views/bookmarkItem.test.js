import React from 'react';
import { shallow } from 'enzyme';
import BookmarkItem from '../../views/BookmarkItem';


describe('<BookmarkItem>', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<BookmarkItem />);
    expect(wrapper).toMatchSnapshot();
  });
});
