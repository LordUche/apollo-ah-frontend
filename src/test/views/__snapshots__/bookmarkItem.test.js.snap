// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookmarkItem> should render without crashing 1`] = `
<div
  className="col-lg-4 col-sm-6"
>
  <Link
    className="link-surface"
    replace={false}
    to="/article/undefined"
  />
  <div
    className="article transition"
  >
    <div
      className="article-cover"
    >
      <img
        alt=""
        className="article-cover-img transition"
      />
    </div>
    <div
      className="article-item"
    >
      <span
        className="article-category"
      />
      <h4 />
      <p />
      <div
        className="article-author"
      >
        <img
          alt=""
        />
        <div>
          <h6 />
          <span
            className="article-author-dot"
          />
          <span />
          <i
            className="fas fa-bookmark transition bookmarked"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
